en/docs/develop/mi-for-vscode/best-practices.md

```diff
@@ -0,0 +1,234 @@
+
+## Mediators & Connectors Best Practices
+
+### General mediator usage
+
+- Add a proper description to each mediator to clarify its purpose and usage within the integration flow.
+- Prefer the `Variable Mediator` over the `Property Mediator` for storing intermediate values during message processing. The `Variable Mediator` is optimized for temporary data storage and reuse within the integration flow, reducing redundant computations and improving performance.
+- Reserve the `Property Mediator` for controlling message flow, such as setting properties that influence routing, error handling, or other flow-specific logic.
+- Use clear, descriptive variable names to enhance the readability and maintainability of your integration logic.
+- If you want to repeatedly use the same mediation sequence, define it as a `Sequence`. You can then invoke this reusable sequence from your main sequence, multiple proxy services, or REST APIs. Use the  `Call Sequence` mediator to call the saved sequence, or select it as the `InSequence` or `FaultSequence` when configuring a proxy service or REST API. This approach promotes modularity, reduces duplication, and simplifies maintenance across your integration flows.
+
+### General connector usage
+
+- Reuse shared connections across multiple integrations whenever possible to optimize resource usage and enhance performance. Connection pooling is managed per connection, so minimizing the number of distinct connections helps reduce overhead and improve scalability.
+- Create separate connections only when required, such as for high-throughput integrations. Dedicated connections provide isolated connection pools, which can be beneficial for integrations that demand high availability.
+
+### Log mediator
+
+- Use Log mediators as checkpoints in message flows to track and debug integration logic. In development, set the log category to `DEBUG` and configure the global log level for `org.apache.synapse.mediators.builtin.LogMediator` accordingly. Include relevant payload information, request metadata (e.g., user ID, timestamp), and unique identifiers (e.g., correlation IDs) to aid in troubleshooting and tracing requests across services.
+- In production, restrict Log mediators to Fault sequences to avoid excessive logging and performance overhead. Always log key error details such as `ERROR_CODE` and use correlation IDs to link related log entries. Place Log mediators within Fault sequences to ensure error information is captured in `wso2carbon.log`. Set the global log level to `INFO` to enable essential troubleshooting without exposing sensitive or verbose information.
+- Avoid logging sensitive information, such as user credentials or personal data, to protect privacy and comply with data protection regulations. Use descriptive messages and implement structured logging to capture key attributes and context information in a consistent format. Regularly review and adjust log levels to balance visibility and performance.
```

Comment:
Let's link https://mi.docs.wso2.com/en/latest/administer/logging-and-monitoring/logging/masking-sensitive-information-in-logs/ here

Reply:
added in [53c308d](https://github.com/wso2/docs-mi/pull/1789/commits/53c308d430f93369483cedbe9dd8f1c8cd66e189)